integration/tools/validation/src/main/java/alluxio/fuse/correctness/ReadValidation.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.fuse.correctness;
+
+import alluxio.fuse.correctness.client.RandomReadClient;
+import alluxio.fuse.correctness.client.SequentialReadClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class validates the read correctness of AlluxioFuse.
+ */
+public class ReadValidation {
+  /**
+   * Starts validating read correctness of AlluxioFuse.
+   *
+   * @param options contains the options for the test
+   */
+  public static void run(Options options) {
+    for (long fileSize: Constants.FILE_SIZES) {
+      System.out.println(String.format(
+          Constants.TESTING_FILE_SIZE_FORMAT, IOOperation.Read, fileSize));
+      List<String> localFileList = new ArrayList<>(options.getNumFiles());
+      List<String> fuseFileList = new ArrayList<>(options.getNumFiles());
+      createTestFiles(fileSize, options, localFileList, fuseFileList);
+      try {
+        for (int bufferSize: Constants.BUFFER_SIZES) {
+          validate(IOOperation.SequentialRead,
```

Comment:
can we avoid creating the number of threads for each file size + buffer size + operation combination?

Reply:
can the whole thread pool be shared by all validation?

Reply:
ok let's use a thread pool